### ISSUE TITLE
[BUG] Jmeter web driver sampler

### ISSUE THREAD

**Issue by ravimainali** [microsoft/azure-load-testing#58]
We want to run Jmeter web driver sampler for real time browser load test so since we dont have control over load engine to copy the web driver exe , therefore we assume that these load engine are one of the Azure Hosted Vm and since Azure hosted vm will by default have chrome and chrome web driver installed so we tried using web driver enviornmnet variable to give the chrome driver path in Jmeter web sampler request but when we run the test from load serive it gives error saying web driver can not found at the specified path so want to know first is it possible or supported to run Jmeter web driver sample in load service and if yes than is there any other way to run web driver sampler othan then what i descriped above ?

AB#1490266

**Comment by ravimainali** [comment]
We have a web application , we want to do a front end web ui load test of this application so we thought of using Jmeter Web driver sampler plugin to do the web ui load test , where this sampler will launch the web browser and then open this web application url followed by navigate to different pages. But Jmeter web driver sampler need a specific web browser web driver as a prerequisite, say for running the load test in chrome browser we need to first copy the chromewebdriver.exe to the particular system where we want to run the test . So i want to know how do we copy this chromedriver exe in this load engine since we dont have any control to this load engine ?

**Comment by kaumishra** [comment]
@ravimainali Thanks for sharing. Let me check on this and get back with an update.

**Comment by kaumishra** [comment]
@ravimainali I checked on this. Currently we do not support uploading executables to our test engines. I have added this to our backlog. I will update this thread once we prioritize it.

**Comment by shobhitkasliwal-onbe** [comment]
@kaumishra  - any update on this? 
I am also looking to test the jmeter webdriver sampler to load test UI.

**Comment by Nagarjuna-Vipparthi** [comment]
@ravimainali @shobhitkasliwal-onbe @SagarBisen34 You can now use Web Driver Sampler with Azure Load Testing. However, owing to its resource intensive nature, it is suggested to have relatively lesser virtual users as compared to other samplers. 
https://azure.microsoft.com/en-us/updates/azure-load-testing-supports-web-driver-sampler-plugin-for-apache-jmeter/ 
https://learn.microsoft.com/azure/load-testing/resource-jmeter-support

Monitor engine health to optimize the utilization of the test engines and modify your script accordingly - https://learn.microsoft.com/azure/load-testing/resource-jmeter-support

**Comment by jchingtaxsystems** [comment]
hello @Nagarjuna-Vipparthi , related to this feature :

Is there way I could push Engines to update its chromedriver.exe version? Currently we're stuck at v128

Sample error from our test:

`Attempt to select document on computation file on large period - Large Period3: Expected condition failed: waiting for visibility of element located by By.xpath: //at-print-preview//h3[text()='Select document'] (tried for 120 second(s) with 500 milliseconds interval)
Build info: version: '4.13.0', revision: 'ba948ece5b*'
System info: os.name: 'Linux', os.arch: 'amd64', os.version: '6.6.112.1-2.azl3', java.version: '21.0.9'
Driver info: org.openqa.selenium.chrome.ChromeDriver
Capabilities {acceptInsecureCerts: false, browserName: chrome, browserVersion: 128.0.6613.84, chrome: {chromedriverVersion: 128.0.6613.84 (606aa55c7d68..., userDataDir: /tmp/.org.chromium.Chromium...}, fedcm:accounts: true, goog:chromeOptions: {debuggerAddress: localhost:25133}, networkConnectionEnabled: false, pageLoadStrategy: normal, platformName: linux, proxy: Proxy(system), se:cdp: ws://localhost:25133/devtoo..., se:cdpVersion: 128.0.6613.84, setWindowRect: true, strictFileInteractability: false, timeouts: {implicit: 0, pageLoad: 300000, script: 30000}, unhandledPromptBehavior: dismiss and notify, webauthn:extension:credBlob: true, webauthn:extension:largeBlob: true, webauthn:extension:minPinLength: true, webauthn:extension:prf: true, webauthn:virtualAuthenticators: true}
Session ID: fee3c9a812da71daf5d46b1134362231`

Basically its an intentional error but it tells us what is the browserVersion being used.

We tried uploading chromedriver.exe (latest) on config.yaml file but still stuck to v128
